### PR TITLE
Add vector tiles to Datasource manager dialog and Layers menu

### PR DESCRIPTION
**Pull request by alexbruy**
## Description
Adds new tab to the Datasource Manager dialog allowing to manage/add vector tile connections, 
including import/export. Also add corresponding entry to Layers menu.

### CODE REVIEW

**Comment by alexbruy** [comment]
Hope @elpaso is a bit more happy now.

**Comment by elpaso** [comment]
I was already happy but I'm happier now :)

File: src/app/qgisapp.cpp [hunk 1]
```diff
@@ -5538,6 +5540,46 @@ QgsMeshLayer *QgisApp::addMeshLayerPrivate( const QString &url, const QString &b
...
+  QgsSettings settings;
+
+  QString base( baseName );
+
+  if ( settings.value( QStringLiteral( "qgis/formatLayerName" ), false ).toBool() )
+  {
+    base = QgsMapLayer::formatLayerName( base );
+  }
+
+  QgsDebugMsg( "completeBaseName: " + base );
```
Comment (nyalldawson) [inline src/app/qgisapp.cpp:5560]: QgsDebugMsgLevel

**Review by nyalldawson — APPROVED** [review]

**Comment by pigreco** [comment]
Please add `GUI/UX` label

**Pull request merged**